var/spack/repos/builtin/packages/py-llvmlite/package.py

```diff
@@ -28,13 +29,14 @@ class PyLlvmlite(PythonPackage):
     version("0.23.0", sha256="bc8b1b46274d05b578fe9e980a6d98fa71c8727f6f9ed31d4d8468dce7aa5762")
 
     depends_on("py-setuptools", type="build")
+    depends_on("python@3.8:", type=("build", "run"), when="@0.40.0:")
     depends_on("python@3.7:3.10", type=("build", "run"), when="@0.38.0:")
```

Comment:
0.40.0 also introduces [python 3.11 support](https://github.com/numba/llvmlite/pull/869), which wasn't there for 0.39:
```suggestion
    depends_on("python@3.8:3.11", type=("build", "run"), when="@0.40.0:")
    depends_on("python@3.7:3.10", type=("build", "run"), when="@0.38.0:0.39")
```